app/questionnaire/router.py

```diff
@@ -89,6 +102,21 @@ def get_first_incomplete_location_in_survey(self):
         last_block_id = self._schema.get_last_block_id_for_section(all_section_ids[-1])
         return Location(block_id=last_block_id)
 
+    def get_first_incomplete_location_for_section(self, section_id, routing_path):
+        if section_id in self._progress_store:
+            for location in routing_path:
+                if location not in self._progress_store.get_completed_locations(
+                    section_id
+                ):
+                    return location
+        return routing_path[0]
+
+    def get_last_complete_location(self, section_id, routing_path):
```

Comment:
This should be named `get_last_complete_location_for_section` to be consistent with `get_first_incomplete_location_for_section`